cmd/commands/datasetPublishData.go

```diff
@@ -1,349 +1,334 @@
+	Args: cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		// const PROD_RSYNC_RETRIEVE_SERVER string = "ebarema4in.psi.ch"
+		// const TEST_RSYNC_RETRIEVE_SERVER string = "ebaremat1in.psi.ch"
+		// const DEV_RSYNC_RETRIEVE_SERVER string = "arematest2in.psi.ch"
+		// ===== local consts =====
+
+		// ===== variables =====
+		var APIServer string = PROD_API_SERVER
+
+		var client = &http.Client{
+			Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: false}},
+			Timeout:   10 * time.Second}
+
+		type PageData struct {
+			Doi           string
+			PageTitle     string
+			BrowseUrls    []string
+			SizeArray     []int
+			NumFilesArray []int
+		}
 
-// const PROD_RSYNC_RETRIEVE_SERVER string = "ebarema4in.psi.ch"
-// const TEST_RSYNC_RETRIEVE_SERVER string = "ebaremat1in.psi.ch"
-// const DEV_RSYNC_RETRIEVE_SERVER string = "arematest2in.psi.ch"
+		// ===== functions =====
+		assembleRsyncCommands := func(datasetDetails []datasetUtils.Dataset) []string {
+			batchCommands := make([]string, 0)
+			for _, dataset := range datasetDetails {
+				shortDatasetId := strings.Split(dataset.Pid, "/")[1]
+				fullDest := "/datasets" + dataset.SourceFolder
+				command := "ssh " + PUBLISHServer + " mkdir -p " + fullDest + ";" +
+					"ssh " + PUBLISHServer + " chown -R egli " + fullDest + ";" +
+					"ssh " + PUBLISHServer + " chmod -R 755 " + fullDest + ";" +
+					"/usr/bin/rsync -av -e ssh " + RETRIEVELocation + shortDatasetId + "/ " + PUBLISHServer + ":" + fullDest
+				batchCommands = append(batchCommands, command)
+			}
+			return batchCommands
+		}
 
-const MANUAL string = "http://melanie.gitpages.psi.ch/SciCatPages/#sec-5"
+		executeCommands := func(batchCommands []string) {
+			log.Printf("\n\n\n====== Starting transfer of dataset files: \n\n")
+			for _, batchCommand := range batchCommands {
+				cmd := exec.Command("/bin/sh", "-c", batchCommand)
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				//log.Printf("Running %v.\n", cmd.Args)
+				log.Printf("\n=== Transfer command: %s.\n", batchCommand)
 
-const APP = "datasetPublishData"
+				err := cmd.Run()
 
-var APIServer string = PROD_API_SERVER
-var RSYNCServer string = PROD_RSYNC_RETRIEVE_SERVER
+				if err != nil {
+					log.Fatal(err)
+				}
+			}
+		}
 
-var client = &http.Client{
-	Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: false}},
-	Timeout:   10 * time.Second}
-var scanner = bufio.NewScanner(os.Stdin)
-var VERSION string
+		createWebpage := func(urls []string, title string, doi string, datasetDetails []datasetUtils.Dataset,
+			publishedDataId string, userpass *string, token *string) {
+			// A Function that returns the longest common prefix path (runes)
+			// from the array of strings
+			commonPrefix := func(arr []string) string {
+				// return shortest string, length given in bytes
+				findMinLength := func(arr []string) int {
```

Comment:
why do these need to become internal functions? is this because the command is now a pointer? I am not a fun of internal functions, as they will be very hard to test later

Reply:
they are internal because the functions are only used once and only exist to make the code more readable. If they weren't internal, the functions' definitions would be shared among all the commands.  This wasn't an issue when we had lots of little "main" packages, but now there's a bigger "cmd" package that ties everything together.